.github/workflows/infra_plan.yaml

```diff
@@ -0,0 +1,162 @@
+    environment: ${{ inputs.environment }}-ci
+    permissions:
+      id-token: write
+      contents: read
+      pull-requests: write
+    env:
+      ARM_CLIENT_ID: ${{ secrets.ARM_CLIENT_ID }}
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+    steps:
+
+      # Set the directory where the Terraform files are located
+      # The directory the value is then available in ${{ steps.directory.outputs.dir }}
+      - name: Set directory
+        id: directory
+        run: |
+          set -e
+          if [ -z "${{ inputs.environment }}" ] || [ -z "${{ inputs.region }}" ]; then
+            echo "Both environment and region must be provided."
+            exit 1
+          else
+            # The directory is expected to be in the format 
+            #  infra/resources/${{ inputs.environment }}/${{ inputs.region }}
+            # Example: infra/resources/prod/westeurope
+            echo "dir=infra/resources/${{ inputs.environment }}/${{ inputs.region }}" >> $GITHUB_OUTPUT
+          fi
+
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        name: Checkout
+
+      - name: Set Environment Variables
+        if: ${{ inputs.env_vars }}
+        run: |
+          for i in "${{ inputs.env_vars }}"
+          do
+            printf "%s\n" $i >> $GITHUB_ENV
+          done
+
+      - name: Azure Login
+        uses: azure/login@v2 # v2.0.0
+        with:
+          client-id: ${{ env.ARM_CLIENT_ID }}
+          tenant-id: ${{ env.ARM_TENANT_ID }}
+          subscription-id: ${{ env.ARM_SUBSCRIPTION_ID }}
+
+      - name: Set Terraform Version
+        id: set-terraform-version
+        run: |
+          echo "terraform_version=$(cat .terraform-version)" >> $GITHUB_OUTPUT
+
+      - uses: hashicorp/setup-terraform@a1502cd9e758c50496cc9ac5308c4843bcd56d36 # v3.0.0
+        name: Setup Terraform
+        with:
+          terraform_version: ${{ steps.set-terraform-version.outputs.terraform_version }}
+
+      - name: Terraform Init
+        working-directory: ${{ steps.directory.outputs.dir }}
+        run: |
+          terraform init
+
+      # Run Terraform Plan
+      # The plan output is saved in a file and then processed to remove unnecessary lines
+      # The step never fails but the result is checked in the next step
+      # This is because we want to post the plan output in the PR even if the plan fails
+      - name: Terraform Plan
+        id: plan
+        working-directory: ${{ steps.directory.outputs.dir }}
+        run: |
+          set -e
+
+          echo "no output" > plan_output_multiline.txt
+          terraform plan -lock-timeout=3000s -no-color | tee plan_output.txt
+          OUTPUT=$(grep -Ev "Refreshing state|state lock|Reading|Read" plan_output.txt | tail -c 60000)
+          echo "$OUTPUT" > plan_output_multiline.txt
+
+          if [ $? -ne 0 ]; then
+            echo "result=failed" >> $GITHUB_OUTPUT
+          fi
+
+      # Post the plan output in the PR
```

Comment:
why don't we put this output into the action summary instead of a comment? ie. 

`>> $GITHUB_STEP_SUMMARY`

Reply:
<img src="https://zuts.files.wordpress.com/2015/06/walter-cool.gif"/>

Reply:
Good idea, we could do both imho (PR comment has always the last output, step summary is the detail of selected run)